plugins/modules/junos_routing_instances.py

```diff
@@ -0,0 +1,765 @@
+          - vrf
+      interfaces:
+        description: Interface name for this routing instance.
+        type: list
+        elements: dict
+        suboptions:
+          name:
+            description: Specify name of the interface.
+            type: str
+          protect_interface:
+            description: Specify name of the protected interface.
+            type: str
+      l2vpn_id:
+        description: Layer-2 vpn-id for this instance.
+        type: str
+      no_irb_layer_2_copy:
+        description: Disable transmission of layer-2 copy of packets of irb routing-interface.
+        type: bool
+      no_normalization:
+         description: Disable vlan id normalization for interfaces.
+         type: bool
+      no_local_switching:
+        description: Disable vlan id normalization for interfaces.
+        type: bool
+      no_vrf_advertise:
+        description: Disable vlan id normalization for interfaces.
+        type: bool
+      no_vrf_propagate_ttl:
+        description: Disable TTL propagation from IP to MPLS (on push) and MPLS to IP (on pop).
+        type: bool
+      qualified_bum_pruning_mode:
+        description: Enable BUM pruning for VPLS instance.
+        type: bool
+      route_distinguisher:
+        description: Route distinguisher for this instance
+        type: str
+      routing_interface:
+        description: Routing interface name for this routing-instance.
+        type: list
+        elements: str
+      vrf_imports:
+        description: Import policy for VRF instance RIBs.
+        type: list
+        elements: str
+      vrf_exports:
+        description: Export policy for VRF instance RIBs.
+        type: list
+        elements: str
+
+  state:
+    description:
+      - The state the configuration should be left in.
+    type: str
+    choices:
+    - merged
+    - replaced
+    - overridden
+    - deleted
+    - parsed
+    - gathered
+    - rendered
+    default: merged
+"""
+EXAMPLES = """
+# Using merged
+#
+# Before state
+# ------------
+#
+# admin# show routing-instances
+#
+# [edit]
+# vagrant@vsrx# show policy-options
+# policy-statement test-policy {
+#     term t1 {
+#         then reject;
+#     }
+# }
+# policy-statement test-policy-1 {
+#     term t1 {
+#         then reject;
+#     }
+# }
+
+- name: Merge Junos BGP address family configuration
+  junipernetworks.junos.junos_routing_instances:
+    config:
+      - name: "test"
+        type: "vrf"
+        route_distinguisher: "10.58.255.1:37"
+        vrf_imports:
+          - "test-policy"
+        vrf_exports:
+          - "test-policy"
+          - "test-policy-1"
+        interfaces:
+          - name: "sp-0/0/0.0"
+          - name: "gr-0/0/0.0"
+        connector_id_advertise: true
+      - name: "forwardinst"
+        type: "forwarding"
+        description: "Configured by Ansible Content Team"
+    state: merged
+#
+# -------------------------
+# Module Execution Result
+# -------------------------
+#
+# After state
```

Comment:
@rohitthakur2590 can you plz include the cmd ran in the examples as well

Reply:
Big netconf based xml trees got generated, if required I can add that in different PR.